router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/configParser.go

```diff
@@ -0,0 +1,26 @@
+package marketobulkupload
+
+import "encoding/json"
+
+func (m *MarketoConfig) UnmarshalJSON(data []byte) error {
+	var intermediate intermediateMarketoConfig
+	if err := json.Unmarshal(data, &intermediate); err != nil {
+		return err
+	}
+
+	m.ClientId = intermediate.ClientId
+	m.ClientSecret = intermediate.ClientSecret
+	m.MunchkinId = intermediate.MunchkinId
+	m.DeduplicationField = intermediate.DeduplicationField
+	m.FieldsMapping = make(map[string]string)
+
+	for _, mapping := range intermediate.ColumnFieldsMapping {
+		from, fromOk := mapping["from"]
+		to, toOk := mapping["to"]
+		if fromOk && toOk {
+			m.FieldsMapping[from] = to
+		}
+	}
+
+	return nil
+}
```

Comment:
Do we need this separate file? Can't we move this function to util?

Reply:
This we will deprecate once we move to vdm hence for now kept separate (this is only needed to parse the ui mapping)